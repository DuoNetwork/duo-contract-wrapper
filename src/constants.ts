export const ETH_MAINNET_ID = 1;
export const ETH_KOVAN_ID = 42;

export const DUMMY_ADDR = '0x0';

export const CTD_INCEPTION = 'Inception';
export const CTD_TRADING = 'Trading';
export const CTD_PRERESET = 'PreReset';
export const CTD_RESET = 'Reset';
export const CTD_LOADING = 'Loading';

export const BTV_UP_RESET = 'UpwardReset';
export const BTV_DOWN_RESET = 'DownwardReset';
export const BTV_PERIOD_RESET = 'PeriodicReset';

export const ESP_NOT_STARTED = 'NotStarted';
export const ESP_MODERATOR = 'Moderator';
export const ESP_CONTRACT = 'Contract';

// wallet and contract
export const INCEPTION_BLK_KOVAN = 9355871;
export const INCEPTION_BLK_MAIN = 0;

// managed
export const EVENT_UPDATE_ROLE_MANAGER = 'UpdateRoleManager';
export const EVENT_UPDATE_OPERATOR = 'UpdateOperator';

// esplanade
export const EVENT_ADD_ADDRESS = 'AddAddress';
export const EVENT_REMOVE_ADDRESS = 'RemoveAddress';
export const EVENT_PROVIDE_ADDRESS = 'ProvideAddress';
export const EVENT_ADD_CUSTODIAN = 'AddCustodian';
export const EVENT_ADD_OTHER_CONTRACT = 'AddOtherContract';
export const EVENT_START_CONTRACT_VOTING = 'StartContractVoting';
export const EVENT_TERMINATE_CONTRACT_VOTING = 'TerminateContractVoting';
export const EVENT_START_MODERATOR_VOTING = 'StartModeratorVoting';
export const EVENT_TERMINATEBY_TIMEOUT = 'TerminateByTimeOut';
export const EVENT_VOTE = 'Vote';
export const EVENT_COMPLETE_VOTING = 'CompleteVoting';
export const EVENT_REPLACE_MODERATOR = 'ReplaceModerator';

// custodian
export const EVENT_START_TRADING = 'StartTrading';
export const EVENT_START_PRE_RESET = 'StartPreReset';
export const EVENT_START_RESET = 'StartReset';
export const EVENT_MATURED = 'Matured'
export const EVENT_ACCEPT_PRICE = 'AcceptPrice';
export const EVENT_CREATE = 'Create';
export const EVENT_REDEEM = 'Redeem';
export const EVENT_TOTAL_SUPPLY = 'TotalSupply';
export const EVENT_TRANSFER = 'Transfer';
export const EVENT_APPROVAL = 'Approval';
export const EVENT_COLLECT_FEE = 'CollectFee';
export const EVENT_UPDATE_ORACLE = 'UpdateOracle';
export const EVENT_UPDATE_FEE_COLLECTOR = 'UpdateFeeCollector';

// beethovan
export const EVENT_SET_VALUE = 'SetValue';

// magi
export const EVENT_COMMIT_PRICE = 'CommitPrice';
export const EVENT_UPDATE_PRICE_FEED = 'UpdatePriceFeed';

export const STATE_INCEPTION = '0';
export const STATE_TRADING = '1';
export const STATE_PRERESET = '2';
export const STATE_RESET = '3';
export const RESET_STATE_UP = '0';
export const RESET_STATE_DOWN = '1';
export const RESET_STATE_PERIOD = '2';
export const VOTING_NOT_STARTED = '0';
export const VOTING_MODERATOR = '1';
export const VOTING_CONTRACT = '2';
export const FN_START_CONTRACT = 'startContract';
export const FN_COMMIT_PRICE = 'commitPrice';
export const SRC_MYETHER = 'myether';
export const SRC_INFURA = 'infura';
export const PROVIDER_LOCAL_HTTP = 'http://localhost:8545';
// export const PROVIDER_LOCAL_WS = 'ws://localhost:8546';
export const PROVIDER_INFURA_MAIN_WS = 'wss://mainnet.infura.io/ws';
export const PROVIDER_INFURA_KOVAN_WS = 'wss://kovan.infura.io/ws';
export const PROVIDER_MYETHER_MAIN = 'https://api.myetherapi.com/eth';
export const PROVIDER_MYETHER_ROP = 'https://api.myetherapi.com/rop';
export const PROVIDER_INFURA_MAIN = 'https://mainnet.infura.io';
export const PROVIDER_INFURA_KOVAN = 'https://kovan.infura.io';
export const TRANSFER_TOKEN_INTERVAL = 30; // in seconds
export const TRANSFER_TOKEN_GAS = 120000;
export const TRANSFER_TOKEN_GAS_TH = 0.01; // in ether
export const SET_VALUE_GAS = 50000;
export const SET_VALUE_GAS_PRICE = 3; // in Gwei
export const COLLECT_FEE_GAS = 40000;
export const COLLECT_FEE_GAS_PRICE = 4; // in Gwei
export const ADD_ADDR_GAS = 40000;
export const ADD_ADDR_GAS_PRICE = 5; // in Gwei
export const REMOVE_ADDR_GAS = 40000;
export const REMOVE_ADDR_GAS_PRICE = 5; // in Gwei
export const UPDATE_ADDR_GAS = 40000;
export const UPDATE_ADDR_GAS_PRICE = 5; // in Gwei
export const TRANSFER_INTERVAL = 10; // in seconds
export const TRANSFER_GAS_TH = 0.01;
export const REDEEM_INTERVAL = 10; // in seconds
export const REDEEM_GAS = 80000;
export const REDEEM_GAS_TH = 0.005;
export const CREATE_INTERVAL = 10; // in seconds
export const CREATE_GAS = 160000;
export const CREATE_GAS_TH = 0.01;
export const DEFAULT_GAS_PRICE = 5;
export const PRE_RESET_GAS_LIMIT = 120000;
export const RESET_GAS_LIMIT = 4000000;
export const EVENT_FETCH_BLOCK_INTERVAL = 100;
export const EVENT_FETCH_TIME_INVERVAL = 600000;

export const LOG_INFO = 'INFO';
export const LOG_DEBUG = 'DEBUG';
export const LOG_ERROR = 'ERROR';
export const LOG_RANKING: { [level: string]: number } = {
	[LOG_ERROR]: 0,
	[LOG_INFO]: 1,
	[LOG_DEBUG]: 2
};

export const BTV_STATE = {
	LAST_OPERATION_TIME: 0,
	OPERATION_COOLDOWN: 1,
	STATE: 2,
	MIN_BALANCE: 3,
	TOTAL_SUPPLYA: 4,
	TOTAL_SUPPLYB: 5,
	ETH_COLLATERAL_INWEI: 6,
	NAVA_INWEI: 7,
	NAVB_INWEI: 8,
	LAST_PRICE_INWEI: 9,
	LAST_PRICETIME_INSECOND: 10,
	RESET_PRICE_INWEI: 11,
	RESET_PRICETIME_INSECOND: 12,
	CREATE_COMMINBP: 13,
	REDEEM_COMMINBP: 14,
	PERIOD: 15,
	MATURITY: 16,
	PRERESET_WAITING_BLOCKS: 17,
	PRICE_FETCH_COOLDOWN: 18,
	NEXT_RESET_ADDR_INDEX: 19,
	TOTAL_USERS: 20,
	FEE_BALANCE_INWEI: 21,
	RESET_STATE: 22,
	ALPHA_INBP: 23,
	BETA_INWEI: 24,
	PERIOD_COUPON_INWEI: 25,
	LIMIT_PERIODIC_INWEI: 26,
	LIMIT_UPPER_INWEI: 27,
	LIMIT_LOWER_INWEI: 28,
	ITERATION_GAS_THRESHOLD: 29
};
