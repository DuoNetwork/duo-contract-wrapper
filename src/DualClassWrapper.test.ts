import * as CST from './constants';
import DualClassWrapper from './DualClassWrapper';

test('convertCustodianState', () => {
	expect(DualClassWrapper.convertCustodianState(CST.STATE_INCEPTION)).toBe(CST.CTD_INCEPTION);
	expect(DualClassWrapper.convertCustodianState(CST.STATE_PRERESET)).toBe(CST.CTD_PRERESET);
	expect(DualClassWrapper.convertCustodianState(CST.STATE_RESET)).toBe(CST.CTD_RESET);
	expect(DualClassWrapper.convertCustodianState(CST.STATE_TRADING)).toBe(CST.CTD_TRADING);
	expect(DualClassWrapper.convertCustodianState(CST.STATE_MATURED)).toBe(CST.CTD_MATURED);
	expect(DualClassWrapper.convertCustodianState('any')).toBe(CST.CTD_LOADING);
});

test('convertCustodianState', () => {
	expect(DualClassWrapper.convertResetState(CST.RESET_STATE_DOWN)).toBe(CST.BTV_DOWN_RESET);
	expect(DualClassWrapper.convertResetState(CST.RESET_STATE_UP)).toBe(CST.BTV_UP_RESET);
	expect(DualClassWrapper.convertResetState(CST.RESET_STATE_PERIOD)).toBe(CST.BTV_PERIOD_RESET);
	expect(DualClassWrapper.convertResetState('any')).toBe('');
});

test('getTokensPerEth', () => {
	expect(
		DualClassWrapper.getTokensPerEth({
			resetPrice: 100,
			beta: 1,
			alpha: 1
		} as any)
	).toMatchSnapshot();
	expect(
		DualClassWrapper.getTokensPerEth({
			resetPrice: 100,
			beta: 0.8,
			alpha: 1
		} as any)
	).toMatchSnapshot();
	expect(
		DualClassWrapper.getTokensPerEth({
			resetPrice: 100,
			beta: 1,
			alpha: 0.5
		} as any)
	).toMatchSnapshot();
	expect(
		DualClassWrapper.getTokensPerEth({
			resetPrice: 100,
			beta: 1,
			alpha: 2
		} as any)
	).toMatchSnapshot();
});

test('getEthWithTokens', () => {
	expect(
		DualClassWrapper.getEthWithTokens(
			{
				resetPrice: 100,
				beta: 1,
				alpha: 1
			} as any,
			100,
			100
		)
	).toMatchSnapshot();
	expect(
		DualClassWrapper.getEthWithTokens(
			{
				resetPrice: 100,
				beta: 1,
				alpha: 1
			} as any,
			100,
			200
		)
	).toMatchSnapshot();
	expect(
		DualClassWrapper.getEthWithTokens(
			{
				resetPrice: 100,
				beta: 0.8,
				alpha: 1
			} as any,
			100,
			100
		)
	).toMatchSnapshot();
	expect(
		DualClassWrapper.getEthWithTokens(
			{
				resetPrice: 100,
				beta: 1,
				alpha: 0.5
			} as any,
			100,
			100
		)
	).toMatchSnapshot();
	expect(
		DualClassWrapper.getEthWithTokens(
			{
				resetPrice: 100,
				beta: 1,
				alpha: 0.5
			} as any,
			100,
			200
		)
	).toMatchSnapshot();
});

test('getTokenInterestOrLeverage', () => {
	expect(
		DualClassWrapper.getTokenInterestOrLeverage(
			{
				periodCoupon: 0.0001,
				period: 3600000
			} as any,
			true,
			true
		)
	).toMatchSnapshot();
	expect(
		DualClassWrapper.getTokenInterestOrLeverage(
			{
				alpha: 1,
				navB: 1.5
			} as any,
			true,
			false
		)
	).toMatchSnapshot();
	expect(
		DualClassWrapper.getTokenInterestOrLeverage(
			{
				navA: 1.2
			} as any,
			false,
			true
		)
	).toMatchSnapshot();
	expect(
		DualClassWrapper.getTokenInterestOrLeverage(
			{
				alpha: 0.5,
				navB: 1.5
			} as any,
			false,
			false
		)
	).toMatchSnapshot();
});

test('calculateNav beethoven', () => {
	expect(
		DualClassWrapper.calculateNav(
			{
				resetPrice: 100,
				resetPriceTime: 0,
				alpha: 1,
				beta: 1,
				period: 5,
				periodCoupon: 0.000001
			} as any,
			true,
			120,
			10
		)
	).toMatchSnapshot();
	expect(
		DualClassWrapper.calculateNav(
			{
				resetPrice: 100,
				resetPriceTime: 0,
				alpha: 1,
				beta: 1,
				period: 5,
				periodCoupon: 0.000001
			} as any,
			true,
			80,
			10
		)
	).toMatchSnapshot();
	expect(
		DualClassWrapper.calculateNav(
			{
				resetPrice: 100,
				resetPriceTime: 0,
				alpha: 1,
				beta: 1,
				period: 5,
				periodCoupon: 0.000001
			} as any,
			true,
			40,
			10
		)
	).toMatchSnapshot();

	expect(
		DualClassWrapper.calculateNav(
			{
				resetPrice: 100,
				resetPriceTime: 0,
				alpha: 2,
				beta: 1,
				period: 5,
				periodCoupon: 0.000001
			} as any,
			true,
			120,
			10
		)
	).toMatchSnapshot();
	expect(
		DualClassWrapper.calculateNav(
			{
				resetPrice: 100,
				resetPriceTime: 0,
				alpha: 2,
				beta: 1,
				period: 5,
				periodCoupon: 0.000001
			} as any,
			true,
			80,
			10
		)
	).toMatchSnapshot();
	expect(
		DualClassWrapper.calculateNav(
			{
				resetPrice: 100,
				resetPriceTime: 0,
				alpha: 2,
				beta: 1,
				period: 5,
				periodCoupon: 0.000001
			} as any,
			true,
			40,
			10
		)
	).toMatchSnapshot();
	expect(
		DualClassWrapper.calculateNav(
			{
				resetPrice: 100,
				resetPriceTime: 0,
				alpha: 0.5,
				beta: 1,
				period: 5,
				periodCoupon: 0.000001
			} as any,
			true,
			120,
			10
		)
	).toMatchSnapshot();
	expect(
		DualClassWrapper.calculateNav(
			{
				resetPrice: 100,
				resetPriceTime: 0,
				alpha: 0.5,
				beta: 1,
				period: 5,
				periodCoupon: 0.000001
			} as any,
			true,
			80,
			10
		)
	).toMatchSnapshot();
	expect(
		DualClassWrapper.calculateNav(
			{
				resetPrice: 100,
				resetPriceTime: 0,
				alpha: 0.5,
				beta: 1,
				period: 5,
				periodCoupon: 0.000001
			} as any,
			true,
			20,
			10
		)
	).toMatchSnapshot();
});

test('calculateNav mozart', () => {
	expect(
		DualClassWrapper.calculateNav(
			{
				resetPrice: 100,
				resetPriceTime: 0,
				alpha: 1,
				beta: 1,
				period: 5,
				periodCoupon: 0.000001
			} as any,
			false,
			120,
			10
		)
	).toMatchSnapshot();
	expect(
		DualClassWrapper.calculateNav(
			{
				resetPrice: 100,
				resetPriceTime: 0,
				alpha: 1,
				beta: 1,
				period: 5,
				periodCoupon: 0.000001
			} as any,
			false,
			80,
			10
		)
	).toMatchSnapshot();
	expect(
		DualClassWrapper.calculateNav(
			{
				resetPrice: 100,
				resetPriceTime: 0,
				alpha: 1,
				beta: 1,
				period: 5,
				periodCoupon: 0.000001
			} as any,
			false,
			40,
			10
		)
	).toMatchSnapshot();

	expect(
		DualClassWrapper.calculateNav(
			{
				resetPrice: 100,
				resetPriceTime: 0,
				alpha: 2,
				beta: 1,
				period: 5,
				periodCoupon: 0.000001
			} as any,
			false,
			120,
			10
		)
	).toMatchSnapshot();
	expect(
		DualClassWrapper.calculateNav(
			{
				resetPrice: 100,
				resetPriceTime: 0,
				alpha: 2,
				beta: 1,
				period: 5,
				periodCoupon: 0.000001
			} as any,
			false,
			80,
			10
		)
	).toMatchSnapshot();
	expect(
		DualClassWrapper.calculateNav(
			{
				resetPrice: 100,
				resetPriceTime: 0,
				alpha: 2,
				beta: 1,
				period: 5,
				periodCoupon: 0.000001
			} as any,
			false,
			40,
			10
		)
	).toMatchSnapshot();
	expect(
		DualClassWrapper.calculateNav(
			{
				resetPrice: 100,
				resetPriceTime: 0,
				alpha: 0.5,
				beta: 1,
				period: 5,
				periodCoupon: 0.000001
			} as any,
			false,
			120,
			10
		)
	).toMatchSnapshot();
	expect(
		DualClassWrapper.calculateNav(
			{
				resetPrice: 100,
				resetPriceTime: 0,
				alpha: 0.5,
				beta: 1,
				period: 5,
				periodCoupon: 0.000001
			} as any,
			false,
			80,
			10
		)
	).toMatchSnapshot();
	expect(
		DualClassWrapper.calculateNav(
			{
				resetPrice: 100,
				resetPriceTime: 0,
				alpha: 0.5,
				beta: 1,
				period: 5,
				periodCoupon: 0.000001
			} as any,
			false,
			20,
			10
		)
	).toMatchSnapshot();
});
