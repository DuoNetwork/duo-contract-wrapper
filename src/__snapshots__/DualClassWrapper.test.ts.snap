// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculateNav beethoven 1`] = `
Array [
  1.000002,
  4.999998,
]
`;

exports[`calculateNav beethoven 2`] = `
Array [
  1.000002,
  1.3999979999999999,
]
`;

exports[`calculateNav beethoven 3`] = `
Array [
  1.000002,
  0.599998,
]
`;

exports[`calculateNav beethoven 4`] = `
Array [
  0.8,
  0,
]
`;

exports[`calculateNav beethoven 5`] = `
Array [
  1.000002,
  1.5999959999999995,
]
`;

exports[`calculateNav beethoven 6`] = `
Array [
  1.000002,
  0.39999600000000024,
]
`;

exports[`calculateNav beethoven 7`] = `
Array [
  0.6000000000000001,
  0,
]
`;

exports[`calculateNav beethoven 8`] = `
Array [
  1.000002,
  1.2999989999999997,
]
`;

exports[`calculateNav beethoven 9`] = `
Array [
  1.000002,
  0.6999990000000001,
]
`;

exports[`calculateNav beethoven 10`] = `
Array [
  0.6000000000000001,
  0,
]
`;

exports[`calculateNav mozart 1`] = `
Array [
  0,
  6,
]
`;

exports[`calculateNav mozart 2`] = `
Array [
  0.8,
  1.5999999999999996,
]
`;

exports[`calculateNav mozart 3`] = `
Array [
  1.2,
  0.40000000000000036,
]
`;

exports[`calculateNav mozart 4`] = `
Array [
  0.8,
  0,
]
`;

exports[`calculateNav mozart 5`] = `
Array [
  0.8,
  2,
]
`;

exports[`calculateNav mozart 6`] = `
Array [
  1.2000000000000002,
  0,
]
`;

exports[`calculateNav mozart 7`] = `
Array [
  0.6000000000000001,
  0,
]
`;

exports[`calculateNav mozart 8`] = `
Array [
  0.8,
  1.4,
]
`;

exports[`calculateNav mozart 9`] = `
Array [
  1.2,
  0.6000000000000001,
]
`;

exports[`calculateNav mozart 10`] = `
Array [
  0.6000000000000001,
  0,
]
`;

exports[`collectFee 1`] = `"Read Only Mode"`;

exports[`collectFee 2`] = `
Array [
  Array [
    10000000000000000000,
  ],
  Array [
    10000000000000000000,
  ],
]
`;

exports[`collectFee 3`] = `
Array [
  Array [
    Object {
      "from": "account",
      "gas": 40000,
      "gasPrice": 1000000000,
    },
  ],
  Array [
    Object {
      "from": "account",
      "gas": 20000,
      "gasPrice": 2000000000,
    },
  ],
]
`;

exports[`create 1`] = `"Read Only Mode"`;

exports[`create 2`] = `"createWithWETHtxHash"`;

exports[`create 3`] = `"createTxHash"`;

exports[`create 4`] = `"createTxHash"`;

exports[`create 5`] = `"createWithWETHtxHash"`;

exports[`create 6`] = `
Array [
  Array [
    1000000000000000000,
    "wethAddr",
  ],
  Array [
    1000000000000000000,
    "wethAddr",
  ],
]
`;

exports[`create 7`] = `
Array [
  Array [
    Object {
      "from": "account",
      "gas": 160000,
      "gasPrice": 1000000000,
      "value": 1000000000000000000,
    },
  ],
  Array [
    Object {
      "from": "account",
      "gas": 20000,
      "gasPrice": 1000000000,
      "value": 1000000000000000000,
    },
  ],
]
`;

exports[`create 8`] = `
Array [
  Array [
    Object {
      "from": "account",
      "gas": 160000,
      "gasPrice": 1000000000,
    },
  ],
  Array [
    Object {
      "from": "account",
      "gas": 20000,
      "gasPrice": 1000000000,
    },
  ],
]
`;

exports[`fetchPrice 1`] = `"wrong env"`;

exports[`fetchPrice 2`] = `
Array [
  Array [
    Object {
      "from": "account",
      "gas": 1000000,
      "gasPrice": 1000000000,
      "nonce": 2,
    },
  ],
  Array [
    Object {
      "from": "account",
      "gas": 20000,
      "gasPrice": 1000000000,
      "nonce": 10,
    },
  ],
]
`;

exports[`getAddress 1`] = `
Object {
  "aToken": "aToken",
  "bToken": "bToken",
  "feeCollector": "feeCollector",
  "operator": "operator",
  "oracle": "oracle",
  "roleManager": "roleManager",
}
`;

exports[`getEthWithTokens 1`] = `2`;

exports[`getEthWithTokens 2`] = `2`;

exports[`getEthWithTokens 3`] = `2.5`;

exports[`getEthWithTokens 4`] = `1.5`;

exports[`getEthWithTokens 5`] = `3`;

exports[`getStates 1`] = `
Object {
  "alpha": 100000000000000,
  "beta": 1,
  "createCommRate": 100000000000000,
  "ethCollateral": 1,
  "feeBalance": 100.00000000000001,
  "iterationGasThreshold": 10000,
  "lastOperationTime": 12345678980000,
  "lastPrice": 1e-9,
  "lastPriceTime": 1e+23,
  "limitLower": 1,
  "limitPeriodic": 15.000000000000002,
  "limitUpper": 2,
  "maturity": 1e+21,
  "minBalance": 100000,
  "navA": 2,
  "navB": 100.00000000000001,
  "nextResetAddrIndex": 10000000000000000000,
  "operationCoolDown": 1e+24,
  "period": 1e+21,
  "periodCoupon": 0.00015000000000000001,
  "preResetWaitingBlocks": 1000000000000000000,
  "priceFetchCoolDown": 1e+24,
  "redeemCommRate": 100000000000000,
  "resetPrice": 10000000.000000002,
  "resetPriceTime": 1e+21,
  "resetState": "",
  "state": "Loading",
  "totalSupplyA": 1e-13,
  "totalSupplyB": 1.0000000000000001e-16,
  "totalUsers": 1e+21,
}
`;

exports[`getTokenInterestOrLeverage 1`] = `3153600.0000000005`;

exports[`getTokenInterestOrLeverage 2`] = `0.8760000000000001`;

exports[`getTokenInterestOrLeverage 3`] = `1.6666666666666667`;

exports[`getTokenInterestOrLeverage 4`] = `-0.6666666666666667`;

exports[`getTokenInterestOrLeverage 5`] = `1.6666666666666667`;

exports[`getTokensPerEth 1`] = `
Array [
  50,
  50,
]
`;

exports[`getTokensPerEth 2`] = `
Array [
  40,
  40,
]
`;

exports[`getTokensPerEth 3`] = `
Array [
  33.333333333333336,
  66.66666666666667,
]
`;

exports[`getTokensPerEth 4`] = `
Array [
  66.66666666666667,
  33.333333333333336,
]
`;

exports[`getUserAddress 1`] = `
Array [
  Array [
    1,
  ],
]
`;

exports[`redeem 1`] = `"Read Only Mode"`;

exports[`redeem 2`] = `"redeemTxHash"`;

exports[`redeem 3`] = `"redeemTxHash"`;

exports[`redeem 4`] = `
Array [
  Array [
    1000000000000000000,
    1000000000000000000,
  ],
  Array [
    1000000000000000000,
    1000000000000000000,
  ],
]
`;

exports[`redeem 5`] = `
Array [
  Array [
    Object {
      "from": "account",
      "gas": 80000,
      "gasPrice": 1000000000,
    },
  ],
  Array [
    Object {
      "from": "account",
      "gas": 20000,
      "gasPrice": 1000000000,
    },
  ],
]
`;

exports[`redeemAll 1`] = `"Read Only Mode"`;

exports[`redeemAll 2`] = `"redeemAllTxHash"`;

exports[`redeemAll 3`] = `"redeemAllTxHash"`;

exports[`redeemAll 4`] = `
Array [
  Array [
    Object {
      "from": "account",
      "gas": 80000,
      "gasPrice": 1000000000,
    },
  ],
  Array [
    Object {
      "from": "account",
      "gas": 20000,
      "gasPrice": 100000,
    },
  ],
]
`;

exports[`setValue 1`] = `"Read Only Mode"`;

exports[`setValue 2`] = `
Array [
  Array [
    1,
    10,
  ],
  Array [
    1,
    10,
  ],
]
`;

exports[`setValue 3`] = `
Array [
  Array [
    Object {
      "from": "account",
      "gas": 50000,
      "gasPrice": 1000000000,
    },
  ],
  Array [
    Object {
      "from": "account",
      "gas": 20000,
      "gasPrice": 2000000000,
    },
  ],
]
`;

exports[`startCustodian 1`] = `"wrong env"`;

exports[`startCustodian 2`] = `
Array [
  Array [
    "aAddr",
    "bAddr",
    "oracleAddr",
  ],
  Array [
    "aAddr",
    "bAddr",
    "oracleAddr",
  ],
]
`;

exports[`startCustodian 3`] = `
Array [
  Array [
    Object {
      "from": "account",
      "gas": 1000000,
      "gasPrice": 1000000000,
    },
  ],
  Array [
    Object {
      "from": "account",
      "gas": 20000,
      "gasPrice": 1000000000,
    },
  ],
]
`;

exports[`triggerPreReset 1`] = `"wrong env"`;

exports[`triggerPreReset 2`] = `
Array [
  Array [
    Object {
      "from": "account",
      "gas": 120000,
      "gasPrice": 1000000000,
    },
  ],
  Array [
    Object {
      "from": "account",
      "gas": 20000,
      "gasPrice": 1000000000,
    },
  ],
]
`;

exports[`triggerReset 1`] = `"wrong env"`;

exports[`triggerReset 2`] = `
Array [
  Array [
    Object {
      "from": "account",
      "gas": 4000000,
      "gasPrice": 1000000000,
    },
  ],
  Array [
    Object {
      "from": "account",
      "gas": 20000,
      "gasPrice": 1000000000,
    },
  ],
]
`;
